cloud_images/centos7/dcos_vol_setup.sh

```diff
@@ -53,7 +53,12 @@ function main {
     echo "Adding entry to fstab: $fstab"
     echo "$fstab" >> /etc/fstab
     echo -n "Mounting: $device to $mount_location"
-    mount "$mount_location"
+    until grep ^$device /etc/mtab > /dev/null; do sleep 1; echo -n .; mount "$mount_location" || :; done
+    if [ "$mount_location" = "/var/log" ]; then
+      echo "Recreating /var/log/journal"
+      mkdir /var/log/journal
+      systemd-tmpfiles --create --prefix /var/log/journal
+    fi
```

Comment:
For all of these reasons, I think it's best to keep `Before=local-fs-pre.target`, and no other dependencies. This target is created by the CentOS installation, it is always present, and it is always the first target that systemd activates (at least the first that we can hook into), which is exactly what we want when formatting partitions. If you haven't already looked at them, see the systems-analyze graphs for an illustration of when dcos_vol_setup.service is active with the dependencies as they are in this request.

So to me it seems that adding additional targets or services to the Before directive would only confuse matters for anyone who ends up looking at this systemd unit in the future. :)

Also, it is very important to be aware that `Before=local-fs-pre.target` _must_ be accompanied by `DefaultDependencies=no`; without this, systemd implicitly assigns `After=basic.target` which is much later in the sequence than we want to be.